ocs/ocp.py

```diff
@@ -134,23 +152,75 @@ def apply(self, yaml_file):
         command = f"apply -f {yaml_file}"
         return self.exec_oc_cmd(command)
 
-    def wait_for_resource_status(
-        self, resource_name, condition, timeout=30, sleep=3
+    def wait_for_resource(
+        self, condition, resource_name='', selector=None, resource_count=0,
+        timeout=60, sleep=3
     ):
         """
         Wait for a resource to reach to a desired condition
 
         Args:
-            resource_name (str): The name of the resource to wait
-                for (e.g.my-pv1)
             condition (str): The desired state the resource should be at
                 This is referring to: status.phase presented in the resource
                 yaml file
                 (e.g. status.phase == Available)
+            resource_name (str): The name of the resource to wait
+                for (e.g.my-pv1)
+            selector (str): The resource selector to search with.
+                Example: 'app=rook-ceph-mds'
+            resource_count (int): How many resources expected to be
+            timeout (int): Time in seconds to wait
+            sleep (int): Sampling time in seconds
+
+        Returns:
+            bool: True in case all resources reached desired condition,
+                False otherwise
+
+        """
+        for sample in TimeoutSampler(
+            timeout, sleep, self.get, resource_name, True, selector
+        ):
+            # Only 1 resource expected to be returned
+            if resource_name:
+                if sample.status.phase == condition:
+                    return True
+            # More than 1 resources returned
+            elif sample.kind == 'List':
+                in_condition = []
+                sample = sample['items']
+                for item in sample:
+                    if item.status.phase == condition:
+                        in_condition.append(item)
+                    if resource_count:
+                        if len(in_condition) == resource_count and (
+                            len(sample) == len(in_condition)
+                        ):
+                            return True
+                    elif len(sample) == len(in_condition):
+                        return True
+
+        return False
+
+    def wait_for_delete(
```

Comment:
the name indicates we wait for the resource to be deleted while the implementation doesn't

Reply:
Fixed,
I added the functionality of waiting for deletion to wait_for_resource so you can specify 'to_delete' = True and the func will wait until no results will exist from the timoutSampler (sample is empty)